src/YEthStakerStrategy.sol

```diff
@@ -147,23 +161,31 @@ contract YEthStakerStrategy is
     function _freeFunds(uint256 _amount) internal override {
         uint256 debt = TokenizedStrategy.totalAssets() -
             asset.balanceOf(address(this));
+
+        // review: why aren't you checking available amount first?
+        // if facility is full, you will have loose assets.
```

Comment:
What do you suggest? Disable withdrawals if the facility is empty?

Reply:
nope, what I am saying, if there is a withdraw of 0.5ETH
you can take it from loose assets.
I believe your math is not taking into account those.

Reply:
If I got you right, you would check loose assets and if there are enough loose assets just skip withdraw?

Base strategy defines that you cannot use loose assets: https://github.com/newmickymousse/yeth-strategy/blob/aa11220dcf911763f83c3f886bb2d3116391d00a/src/YEthStakerStrategy.sol#L129-L130 & https://github.com/newmickymousse/yeth-strategy/blob/aa11220dcf911763f83c3f886bb2d3116391d00a/src/YEthStakerStrategy.sol#L137-L138